src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -28928,6 +28928,244 @@ public void Init(T t)
                 Diagnostic(ErrorCode.WRN_MemberNotNull, "}").WithArguments("field").WithLocation(18, 5));
         }
 
+        [Fact, WorkItem(48996, "https://github.com/dotnet/roslyn/issues/48996")]
+        public void DisallowNullProperty_Constructor()
+        {
+            var source = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+public class C1
+{
+    [DisallowNull, NotNull] public string? F { get; set; }
+
+    public C1() { } // 1
+    public C1(string? F) // 2
+    {
+        this.F = F; // 3
+    }
+
+    public C1(int i)
+    {
+        this.F = i.ToString();
+    }
+}
+
+public class C2
+{
+    [NotNull] public string? F { get; set; }
+
+    public C2() { }
+    public C2(string? F)
+    {
+        this.F = F;
+    }
+
+    public C2(int i)
+    {
+        this.F = i.ToString();
+    }
+}
+
+public class C3
+{
+    [DisallowNull] public string? F { get; set; }
+
+    public C3() { }
+    public C3(string? F)
+    {
+        this.F = F; // 1
+    }
+
+    public C3(int i)
+    {
+        this.F = i.ToString();
```

Comment:
Why does this not warn given that `ToString` has a `string?` return type?

Given that `ToString` can return `string?` or `string~` given the reference assemblies we use I worry that it makes a poor method to use in testing. Developers reading the change will likely have the expectation that the return is `string?` but in our tests it can be either. Think we should use a method that has an non-contextual nullable return in our tests.

Reply:
Changing to use a literal string to improve clarity.